api_v3/services/LiveStreamService.php

```diff
@@ -577,4 +577,102 @@ public function regenerateStreamTokenAction($entryId)
 
 	}
 
+    /**
+     * Archive a live entry which was recorded
+     *
+     * @action archive
+     * @param string $liveEntryId
+     * @param KalturaLiveEntryArchiveOptions $archiveOptions
+     * @return bool
+     * @throws KalturaAPIException
+     * @throws KalturaClientException
+     * @throws PropelException
+     */
+	public function archiveAction($liveEntryId, KalturaLiveEntryArchiveOptions $archiveOptions)
+    {
+        $liveEntry = entryPeer::retrieveByPK($liveEntryId);
+        /** @var LiveStreamEntry $liveEntry */
+        if (!$liveEntry)
+        {
+            KalturaLog::err("Live entry id [" . $liveEntryId . "] not found");
+            throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $liveEntryId);
+        }
+        $cuePoints = CuePointPeer::retrieveByEntryId($liveEntryId);
+
+        $shouldBeDeletedCuePoints = array();
+
+        $notDeletedCuePointTagsArray = explode(",", $archiveOptions->notDeletedCuePointTags);
+        foreach ($cuePoints as $cuePoint)
+        {
+            /** @var CuePoint $cuePoint */
+            KalturaLog::info("Iterating over cue point [" . $cuePoint->getId() . "]");
+            $cuePointTags = explode(',', $cuePoint->getTags());
+            $containsImportantTag = false;
+            foreach ($notDeletedCuePointTagsArray as $importantCuePointTag)
+            {
+                if (in_array($importantCuePointTag, $cuePointTags))
+                {
+                    $containsImportantTag = true;
+                    break;
+                }
+            }
+            if (!$containsImportantTag)
+            {
+                $shouldBeDeletedCuePoints[] = $cuePoint->getId();
+            }
+        }
+
+        CuePointPeer::setCuePointsStatus($shouldBeDeletedCuePoints, CuePointStatus::DELETED);
+
+        $vodEntry = entryPeer::retrieveByPK($liveEntry->getRecordedEntryId());
```

Comment:
Better check if recorded entry id exists and change the logic a bit to match this check.